plugin/pulpcore/plugin/stages/models.py

```diff
@@ -0,0 +1,75 @@
+from gettext import gettext as _
+
+
+class DeclarativeArtifact:
+    """
+    Relates an :class:`~pulpcore.plugin.models.Artifact`, how to download it, and its
+    `relative_path` used later during publishing.
+
+    This is used by the Stages API stages to determine if an
+    :class:`~pulpcore.plugin.models.Artifact` is already present and ensure Pulp can download it in
+    the future. The `artifact` can be either saved or unsaved. If unsaved, the `artifact` attributes
+    may be incomplete because not all digest information can be computed until the
+    :class:`~pulpcore.plugin.models.Artifact` is downloaded.
+
+    Attributes:
+        artifact (:class:`~pulpcore.plugin.models.Artifact`): An
+            :class:`~pulpcore.plugin.models.Artifact` either saved or unsaved. If unsaved, it
+            may have partial digest information attached to it.
+        url (str): the url to fetch the :class:`~pulpcore.plugin.models.Artifact` from.
+        relative_path (str): the relative_path this :class:`~pulpcore.plugin.models.Artifact`
+            should be published at for any Publication.
+        remote (:class:`~pulpcore.plugin.models.Remote`): The remote used to fetch this
```

Comment:
Just curious: is the `:class:` necessary?  The docs seems to generate fine without it.

Reply:
It won't be clickable without this label. All of these class renders are clickable.

Reply:
Ah.